src/bio/bio_buffer.c

```diff
@@ -1188,11 +1188,13 @@ nvme_rw(struct bio_desc *biod, struct bio_rsrvd_region *rg)
 					   page2io_unit(biod->bd_ctxt, pg_idx, BIO_DMA_PAGE_SZ),
 					   page2io_unit(biod->bd_ctxt, rw_cnt, BIO_DMA_PAGE_SZ),
 					   rw_completion, biod);
-		else
+		else {
 			spdk_blob_io_read(blob, channel, payload,
 					  page2io_unit(biod->bd_ctxt, pg_idx, BIO_DMA_PAGE_SZ),
 					  page2io_unit(biod->bd_ctxt, rw_cnt, BIO_DMA_PAGE_SZ),
 					  rw_completion, biod);
+			VALGRIND_MAKE_MEM_DEFINED(payload, rw_cnt * BIO_DMA_PAGE_SZ);
```

Comment:
This should be called only when "if (DAOS_ON_VALGRIND)".

Reply:
Right. Plz see the updated patch.